src/audio/module_adapter/module_adapter.c

```diff
@@ -58,6 +59,10 @@ struct comp_dev *module_adapter_new(const struct comp_driver *drv,
 		break;
 	}
 	}
+#else
+	size = MAILBOX_HOSTBOX_SIZE;
```

Comment:
you mean add a new param ID for getting the IPC payload size for each module? I'd like to keep the IPC type check. For ipc3, we never had the concept of param ID's and I do not think it is a good idea to introduce it now.

Reply:
I prefer to let each module to build config by itself. I don't like switch here

Reply:
Allocating 8K for each module when they might only need few bytes is a bit of waste, I agree with @RanderWang

Reply:
> @ranj063, the firmware crash is happening in a different place now with this set. dmesg
> 
> ```
> [ 2757.709908] snd_sof:sof_ipc4_route_setup: sof-audio-pci-intel-tgl 0000:00:1f.3: bind copier.host.1.1 -> gain.1.1
> [ 2757.709914] snd_sof:sof_ipc4_log_header: sof-audio-pci-intel-tgl 0000:00:1f.3: ipc tx      : 0x45000004|0x6: MOD_BIND
> [ 2757.710024] snd_sof:sof_ipc4_log_header: sof-audio-pci-intel-tgl 0000:00:1f.3: ipc tx reply: 0x65000000|0x6: MOD_BIND
> [ 2757.710051] snd_sof:sof_ipc4_log_header: sof-audio-pci-intel-tgl 0000:00:1f.3: ipc tx done : 0x45000004|0x6: MOD_BIND
> [ 2757.710063] snd_sof:sof_ipc4_route_setup: sof-audio-pci-intel-tgl 0000:00:1f.3: bind gain.1.1 -> mixin.1.1
> [ 2757.710070] snd_sof:sof_ipc4_log_header: sof-audio-pci-intel-tgl 0000:00:1f.3: ipc tx      : 0x45000006|0x2: MOD_BIND
> [ 2757.710202] snd_sof:sof_ipc4_log_header: sof-audio-pci-intel-tgl 0000:00:1f.3: ipc tx reply: 0x65000000|0x2: MOD_BIND
> [ 2757.710222] snd_sof:sof_ipc4_log_header: sof-audio-pci-intel-tgl 0000:00:1f.3: ipc tx done : 0x45000006|0x2: MOD_BIND
> [ 2757.710228] snd_sof:sof_ipc4_route_setup: sof-audio-pci-intel-tgl 0000:00:1f.3: bind mixin.1.1 -> mixout.2.1
> [ 2757.710233] snd_sof:sof_ipc4_log_header: sof-audio-pci-intel-tgl 0000:00:1f.3: ipc tx      : 0x45000002|0x3: MOD_BIND
> [ 2758.213171] sof-audio-pci-intel-tgl 0000:00:1f.3: ipc timed out for 0x45000002|0x3
> [ 2758.213177] sof-audio-pci-intel-tgl 0000:00:1f.3: sof_ipc4_route_setup: failed to bind modules mixin.1.1 -> mixout.2.1
> ```
> 
> fw log:
> 
> ```
> [00:00:00.165,000] <inf> module_adapter: module_init() start
> [00:00:00.165,000] <inf> module_adapter: module_init() done
> [00:00:00.166,000] <inf> pipe: pipeline new pipe_id 2 priority 0
> [00:00:00.166,000] <inf> module_adapter: module_init() start
> [00:00:00.166,000] <inf> module_adapter: module_init() done
> [00:00:00.166,000] <inf> ipc: buffer new size 0x300 id 2.0 flags 0x0
> [00:00:00.166,000] <inf> hda_dai: hda_probe()
> [00:00:00.166,000] <inf> dai: dai_get type 3 index 0 new sref 1
> [00:00:00.166,000] <inf> hda_dma: hda-dmac :7 -> probe
> [00:00:00.167,000] <inf> dma: dma_get() ID 7 sref = 1 busy channels 0
> --- 1 messages dropped ---
> [00:00:00.167,000] <inf> pipe: connect buffer 0 as sink
> --- 2 messages dropped ---
> [00:00:00.167,000] <inf> pipe: connect buffer 0 as source
> [00:00:00.167,000] <inf> ipc: buffer new size 0x300 id 1.0 flags 0x0
> [00:00:00.167,000] <inf> pipe: connect buffer 0 as sink
> [00:00:00.167,000] <inf> pipe: connect buffer 0 as source
> [00:00:00.167,000] <err> os:  ** FATAL EXCEPTION
> [00:00:00.167,000] <err> os:  ** CPU 0 EXCCAUSE 12 (instr PIF data error)
> [00:00:00.167,000] <err> os:  **  PC (nil) VADDR (nil)
> [00:00:00.167,000] <err> os:  **  PS 0x60120
> [00:00:00.167,000] <err> os:  **    (INTLEVEL:0 EXCM: 0 UM:1 RING:0 WOE:1 OWB:1 CALLINC:2)
> [00:00:00.167,000] <err> os:  **  A0 0xbe023178  SP 0x9e066279  A2 0x9e06a4c0  A3 0xbe061910
> [00:00:00.167,000] <err> os:  **  A4 (nil)  A5 0x9e06a740  A6 (nil)  A7 0xbe061870
> [00:00:00.167,000] <err> os:  **  A8 0xbe023adf  A9 0x1 A10 0x9e06a4c0 A11 0x63
> [00:00:00.167,000] <err> os:  ** A12 0x1 A13 0x1 A14 (nil) A15 0xbe06189c
> [00:00:00.167,000] <err> os:  ** LBEG 0xbe03d388 LEND 0xbe03d390 LCOUNT (nil)
> [00:00:00.167,000] <err> os:  ** SAR 0x4
> 
> Backtrace:0xFFFFFFFD:0xBE061870  |<-CORRUPTED
> 
> [00:00:00.167,000] <err> os: >>> ZEPHYR FATAL ERROR 0: CPU exception on CPU 0
> [00:00:00.167,000] <err> os: Current thread: 0x9e066928 (unknown)
> [00:00:00.168,000] <err> os: Halting system
> ```

@ujfalusi this is because we need to add the get_large_cinfignfornmixin/mixoutnas well

Reply:
I agree that 8k is wasteful but maybe we don't even need to copy the mainbox here and pass the pointer directly. Let me update it in the next version